src/core/core.cpp

```diff
@@ -792,10 +793,17 @@ void Core::setUsername(const QString& username)
 
 void Core::setAvatar(const QByteArray& data)
 {
-    QPixmap pic;
-    pic.loadFromData(data);
-    Settings::getInstance().saveAvatar(pic, getSelfId().toString());
-    emit selfAvatarChanged(pic);
+    if (data != nullptr)
+    {
+        QPixmap pic;
+        pic.loadFromData(data);
+        Settings::getInstance().saveAvatar(pic, getSelfId().toString());
+        emit selfAvatarChanged(pic);
+    }
+    else
+    {
+        emit selfAvatarChanged(QPixmap(":/img/contact_dark.svg"));
```

Comment:
I'm not sure about this part. This is emitting qtox default avatar as users empty avatar, that means empty avatar for qtox will be different than empty avatar for µtox/another client. SO empty avatar might in some situation look like custom one.